evadb/optimizer/statement_to_opr_converter.py

```diff
@@ -313,13 +319,85 @@ def visit_explain(self, statement: ExplainStatement):
         self._plan = explain_opr
 
     def visit_create_index(self, statement: CreateIndexStatement):
+        # convert it into a following plan
+        # LogicalCreateIndex
+        #        |
+        #    LogicalProject
+        #        |
+        #    LogicalGet
+
+        table_ref = statement.table_ref
+        catalog_entry = table_ref.table.table_obj
+        logical_get = LogicalGet(table_ref, catalog_entry, table_ref.alias)
+        project_exprs = statement.col_list
+        # if there is a function expr, make col as its children
+        if statement.udf_func:
+            project_exprs = [statement.udf_func.copy()]
+
+        # We need to also store the primary keys of the table in the index to later do
+        # the mapping. Typically, the vector indexes support setting an integer ID with
+        # the embedding. Unfortunately, we cannot support multi-column primary keys for
+        # cases like video table and document table.
+        # Hack: In such cases, we convert them into a unique ID using the following
+        # logic: We assume that the maximum number of files in the table is <=
+        # MAGIC_NUMBER and the number of frames or chunks for each video/document is <=
+        # MAGIC_NUMBER. Based on this assumption, we can safely say that
+        # `_row_id` * MAGIC_NUMBER + `chunk_id` for document table and
```

Comment:
How do you determine the magic number? And when will be the chunking be done?

Reply:
Right now, I set it to an arbitrarily large value. The idea is that as long as the number of files in the table is fewer than this number, our assumption will hold. 

Reg chunking: It is done when we read the document similar to frame decoding in videos.

Reply:
I see. And for video, each video will get a unique ID and each frame will be assigned a different frame ID? Is this assumption correct?

Reply:
We already have these IDs (`_row_id` for video and `id` for frame). Building on top of it. The assumption is these ids won't change across runs. `_row_id` is persisted, so no issue there. `id` is generated at runtime, and as long as the `reader` is deterministic across runs, we don't have a problem.